src/server/status_server.rs

```diff
@@ -48,19 +52,92 @@ impl StatusServer {
         let builder = Server::try_bind(&addr)?;
 
         // Create a status service.
-        let service = |req: Request<Body>| -> FutureResult<Response<Body>, hyper::Error> {
-            let mut response = Response::new(Body::empty());
-
+        let service = |req: Request<Body>| -> Box<dyn Future<Item=Response<Body>, Error=hyper::Error> + Send> {
             match (req.method(), req.uri().path()) {
                 (&Method::GET, "/metrics") => {
-                    *response.body_mut() = Body::from(dump());
+                    let response = Response::builder().body(Body::from(dump())).unwrap();
+                    Box::new(ok(response))
+                }
+                (&Method::GET, "/jeprof") => {
```

Comment:
please abstract to a profile function

Reply:
Ok. I have abstracted them into two profile functions. One for dump profile asynchronously and the other is used for generate response from the result. `async_dump_profile` function can be placed in `tikv_alloc` crate but a timer is needed (and will add a lot of dependencies in `tikv_alloc`), so I don't do this.